plugins/modules/ovirt_disk.py

```diff
@@ -420,118 +429,175 @@ def create_transfer_connection(module, transfer, context, connect_timeout=10, re
-            pos = 0
-            while pos < image_size:
-                to_read = min(image_size - pos, BUF_SIZE)
-                chunk = r.read(to_read)
-                if not chunk:
-                    raise RuntimeError("Socket disconnected")
-                mydisk.write(chunk)
-                pos += len(chunk)
-
-    return transfer(
-        connection,
-        module,
-        otypes.ImageTransferDirection.DOWNLOAD,
-        transfer_func=_transfer,
-    )
+    transfers_service = connection.system_service().image_transfers_service()
+    hosts_service = connection.system_service().hosts_service()
+    transfer = start_transfer(connection, module, otypes.ImageTransferDirection.DOWNLOAD)
+    try:
+        extra_args = {}
+        parameters = inspect.signature(client.download).parameters
+        if "proxy_url" in parameters:
+            extra_args["proxy_url"] = transfer.proxy_url
+        if module.params.get('max_workers') and "max_workers" in parameters:
+            extra_args["max_workers"] = module.params.get('max_workers')
+        client.download(
+            transfer.transfer_url,
+            module.params.get('download_image_path'),
+            module.params.get('auth').get('ca_file'),
+            fmt='qcow2' if module.params.get('format') == 'cow' else 'raw',
+            secure=not module.params.get('auth').get('insecure'),
+            buffer_size=client.BUFFER_SIZE,
+            **extra_args
+        )
+    except Exception as e:
+        cancel_transfer(connection, transfer)
+        raise e
+    finalize_transfer(connection, module, transfer)
+    return True
 
 
 def upload_disk_image(connection, module):
-    def _transfer(transfer_service, transfer_connection, transfer_url):
-        BUF_SIZE = 128 * 1024
-        path = module.params['upload_image_path']
-
-        image_size = os.path.getsize(path)
-        transfer_connection.putrequest("PUT", transfer_url.path)
-        transfer_connection.putheader('Content-Length', "%d" % (image_size,))
-        transfer_connection.endheaders()
-        with open(path, "rb") as disk:
-            pos = 0
-            while pos < image_size:
-                to_read = min(image_size - pos, BUF_SIZE)
-                chunk = disk.read(to_read)
-                if not chunk:
-                    transfer_service.pause()
-                    raise RuntimeError("Unexpected end of file at pos=%d" % pos)
-                transfer_connection.send(chunk)
-                pos += len(chunk)
-
-    return transfer(
-        connection,
-        module,
-        otypes.ImageTransferDirection.UPLOAD,
-        transfer_func=_transfer,
-    )
+    transfers_service = connection.system_service().image_transfers_service()
+    hosts_service = connection.system_service().hosts_service()
+    transfer = start_transfer(connection, module, otypes.ImageTransferDirection.UPLOAD)
+    try:
+        extra_args = {}
+        parameters = inspect.signature(client.download).parameters
```

Comment:
1) Shouldn't it be "client.upload"?
2) What about "max_workers" in case of upload, is it relevant for download only?
3) Order of the first 2 parameters of  "client.download" vs. "client.upload", better be consistent.
4) "fmt" is not relevant for "client.upload()"?

Reply:
fixed the inspect.signature
and added `max_workers` to upload, from what I understood it is also relevant to the client upload

Reply:
Can be marked as resolved.